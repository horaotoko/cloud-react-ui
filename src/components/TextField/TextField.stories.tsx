import { text, boolean } from "@storybook/addon-knobs";
import { TextField as TextFieldComponent } from "./TextField";

export default {
  component: TextFieldComponent,
  title: "components/Text Field",
};

export const TextField = (): JSX.Element => (
  <TextFieldComponent
    placeholder={text("placeholder", "")}
    error={boolean("error", false)}
  />
);
