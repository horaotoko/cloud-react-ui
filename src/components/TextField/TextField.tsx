import { forwardRef } from 'react';
import styled, { css } from 'styled-components';
import { defaultProps } from '../../theme';

export type Props = React.InputHTMLAttributes<HTMLInputElement> & {
  error?: boolean;
};

const StyledInput = styled.input<Props>`
  ${({ theme: { textField }, error = false }) => {
    const colorType = error ? 'error' : 'default';

    return css`
      width: ${textField.width};
      height: ${textField.height};
      padding: ${textField.padding};
      background-color: ${textField.backgroundColor[colorType]};
      border-radius: ${textField.borderRadius};
      border-width: ${textField.borderWidth};
      border-style: ${textField.borderStyle};
      border-color: ${textField.borderColor[colorType]};
      color: ${textField.color};
      font-size: ${textField.fontSize};
      box-sizing: border-box;

      ::placeholder {
        color: ${textField.placeholderColor};
      }
    `;
  }}
`;
StyledInput.defaultProps = defaultProps;

const TextField = forwardRef<HTMLInputElement, Props>(
  ({ type = 'text', ...rest }, ref) => (
    <StyledInput type={type} ref={ref} {...rest} />
  )
);

TextField.displayName = 'TextField';

export { TextField };
