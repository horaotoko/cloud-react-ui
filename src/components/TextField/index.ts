export { TextFieldContainer as TextField } from "./TextFieldContainer";
export type { TextFieldProps } from "./TextFieldContainer";
