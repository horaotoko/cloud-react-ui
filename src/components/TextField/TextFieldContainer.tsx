import { forwardRef } from "react";
import { TextFieldPresenter } from "./TextFieldPresenter";

export type TextFieldProps = React.InputHTMLAttributes<HTMLInputElement> & {
  error?: boolean;
};

export const TextFieldContainer = forwardRef<HTMLInputElement, TextFieldProps>(
  function TextFieldContainer(props, ref) {
    return <TextFieldPresenter ref={ref} {...props} />;
  }
);
