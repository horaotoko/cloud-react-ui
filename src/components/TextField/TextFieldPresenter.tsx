import { forwardRef } from "react";
import styled, { css } from "styled-components";
import { TextFieldProps } from "./TextFieldContainer";
import { defaultProps } from "../../theme";

const StyledInput = styled.input<TextFieldProps>`
  ${({ theme: { textField }, error = false }) => {
    const colorType = error ? "error" : "default";

    return css`
      width: ${textField.width};
      height: ${textField.height};
      padding: ${textField.padding};
      background-color: ${textField.backgroundColor[colorType]};
      border-radius: ${textField.borderRadius};
      border-width: ${textField.borderWidth};
      border-style: ${textField.borderStyle};
      border-color: ${textField.borderColor[colorType]};
      color: ${textField.color};
      font-size: ${textField.fontSize};

      ::placeholder {
        color: ${textField.placeholderColor};
      }
    `;
  }}
`;
StyledInput.defaultProps = defaultProps;

export const TextFieldPresenter = forwardRef<HTMLInputElement, TextFieldProps>(
  function TextFieldPresenter({ type = "text", ...rest }, ref) {
    return <StyledInput type={type} ref={ref} {...rest} />;
  }
);
