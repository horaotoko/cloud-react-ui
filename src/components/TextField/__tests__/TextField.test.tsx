import { render } from "@testing-library/react";
import { TextField } from "../TextField";

describe("TextField", () => {
  it("default", () => {
    const { asFragment, getByRole } = render(<TextField />);
    expect(asFragment()).toMatchSnapshot();

    expect(getByRole("textbox")).toHaveStyle("background-color: #FFFFFF");
    expect(getByRole("textbox")).toHaveStyle("border-color: #D4D8DD");
  });

  it("error props", () => {
    const { getByRole } = render(<TextField error={true} />);

    expect(getByRole("textbox")).toHaveStyle("background-color: #FFEEEB;");
    expect(getByRole("textbox")).toHaveStyle("border-color: #F57575;");
  });
});
