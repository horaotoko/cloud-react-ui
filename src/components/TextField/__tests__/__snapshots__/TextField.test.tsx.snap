// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextField default 1`] = `
<DocumentFragment>
  .c0 {
  width: 100%;
  height: 34px;
  padding: 10px 5px;
  background-color: #FFFFFF;
  border-radius: 4px;
  border-width: 1px;
  border-style: solid;
  border-color: #D4D8DD;
  color: #333333;
  font-size: 13px;
  box-sizing: border-box;
}

.c0::-webkit-input-placeholder {
  color: #999999;
}

.c0::-moz-placeholder {
  color: #999999;
}

.c0:-ms-input-placeholder {
  color: #999999;
}

.c0::placeholder {
  color: #999999;
}

<input
    class="c0"
    type="text"
  />
</DocumentFragment>
`;
